tests/chainer_tests/test_function_node.py

```diff
@@ -471,4 +471,147 @@ def test_retain_outputs_f1(self):
                                          self.f1_output_data[1])
 
 
+def _get_value(x):
+    if isinstance(x, chainer.Variable):
+        return x.data
+    return x
+
+
+class GradTestBase(object):
+
+    shape = 3,
+    x_names = ()
+    y_names = ()
+
+    def _init_attrs(self, names):
+        ret = []
+        for name in names:
+            v = chainer.Variable(
+                numpy.random.randint(-4, 6, self.shape).astype('f'), name=name)
+            ret.append(v)
+            setattr(self, name, v)
+        return ret
+
+    @staticmethod
+    def _get_value(x):
+        if isinstance(x, chainer.Variable):
+            return x.data
+        return x
+
+    @staticmethod
+    def _to_grad_names(names):
+        return ['g%s' % name for name in names]
+
+    def setUp(self):
+        self.xs = self._init_attrs(self.x_names)
+        self.gxs = self._init_attrs(self._to_grad_names(self.x_names))
+        self.gys = self._init_attrs(self._to_grad_names(self.y_names))
+
+    def use_gpu(self):
+        for value in six.itervalues(self.__dict__):
+            if isinstance(value, chainer.Variable):
+                value.to_gpu()
+
+    def forward(self):
+        raise NotImplementedError
+
+    def expected_grad(self):
+        raise NotImplementedError
+
+    def expected_double_grad(self):
+        raise NotImplementedError
+
+    def _print_variables(self, name, vs):
+        print('{}: '.format(name), end='')
```

Comment:
Please add `from __future__ import print_function` to use `end` argument for `print` in Python 2